htlcswitch/link_test.go

```diff
@@ -670,3 +671,133 @@ func TestChannelLinkSingleHopMessageOrdering(t *testing.T) {
 		t.Fatalf("unable to make the payment: %v", err)
 	}
 }
+
+// TestChannelLinkFeeUpdate tests that a channel fee update by the
+// channel link is applied correctly.
+func TestChannelLinkFeeUpdate(t *testing.T) {
```

Comment:
I think that I might know better way of testing the update fee message, in current approach we:
- create additional `mockPeer` structure which is already exists - `mockServer`.
- we manually create the update htlc message and applying this change to the channel.

The test framework which resides in the `test_util.go` have been created to test that two links are able to interact which is each other without being desychronized.

What we could do is create the standard cluster, as we do it in other tests, and trigger update fee handler on `alice` side, with this the update fee message will be sent to the `bob`. After they exchange the fee update messages, as you do in this test you could use standard `makePayment` function to send the payment and assure that state haven't been desynchronised. The part of the test which checks the exact commit fee from my POW should be moved in `channel_test.go`.